framework/src/main/java/org/checkerframework/common/wholeprograminference/scenelib/ASceneWrapper.java

```diff
@@ -0,0 +1,246 @@
+ */
+public class ASceneWrapper {
+
+    /** The AScene being wrapped. */
+    private AScene theScene;
+
+    /** The classes in the scene. */
+    private Map<@BinaryName String, AClassWrapper> classes = new HashMap<>();
+
+    /**
+     * Constructor. Pass the AScene to wrap.
+     *
+     * @param theScene the scene to wrap
+     */
+    public ASceneWrapper(AScene theScene) {
+        this.theScene = theScene;
+    }
+
+    /**
+     * Fetch the classes in this scene, represented as AClassWrapper objects.
+     *
+     * @return an immutable map from binary names to AClassWrapper objects
+     */
+    public Map<@BinaryName String, AClassWrapper> getClasses() {
+        return ImmutableMap.copyOf(classes);
+    }
+
+    /**
+     * Removes all annotations that should be ignored from an AScene.
+     *
+     * @param scene the scene from which to remove annotations
+     * @param annosToIgnore maps the toString() representation of an ATypeElement and its
+     *     TypeUseLocation to a set of names of annotations that should not be added to .jaif files
+     *     for that location.
+     */
+    private void removeIgnoredAnnosFromScene(
+            AScene scene, Map<Pair<String, TypeUseLocation>, Set<String>> annosToIgnore) {
+        for (AClass aclass : scene.classes.values()) {
+            for (AField field : aclass.fields.values()) {
+                removeIgnoredAnnosFromATypeElement(
+                        field.type, TypeUseLocation.FIELD, annosToIgnore);
+            }
+            for (AMethod method : aclass.methods.values()) {
+                // Return type
+                removeIgnoredAnnosFromATypeElement(
+                        method.returnType, TypeUseLocation.RETURN, annosToIgnore);
+                // Receiver type
+                removeIgnoredAnnosFromATypeElement(
+                        method.receiver.type, TypeUseLocation.RECEIVER, annosToIgnore);
+                // Parameter types
+                for (AField param : method.parameters.values()) {
+                    removeIgnoredAnnosFromATypeElement(
+                            param.type, TypeUseLocation.PARAMETER, annosToIgnore);
+                }
+            }
+        }
+    }
+
+    /**
+     * Removes all annotations that should be ignored from an ATypeElement.
+     *
+     * @param typeElt the type element from which to remove annotations
+     * @param loc the location where typeElt in used
+     * @param annosToIgnore maps the toString() representation of an ATypeElement and its
+     *     TypeUseLocation to a set of names of annotations that should not be added to .jaif files
+     *     for that location.
+     */
+    private void removeIgnoredAnnosFromATypeElement(
+            ATypeElement typeElt,
+            TypeUseLocation loc,
+            Map<Pair<String, TypeUseLocation>, Set<String>> annosToIgnore) {
+        String firstKey = typeElt.description.toString() + typeElt.tlAnnotationsHere;
```

Comment:
I'm not sure why this is called `firstKey`; I don't see another one.
More importantly, this seems inconsistent with the `@param` documentation.  That says it's the `toString()` representation, so why not just call `toString()` here?  I suspect either the code or the documentation needs to be updated.